pkg/mod/fs_test.go

```diff
@@ -0,0 +1,48 @@
+package mod
+
+import (
+	"fmt"
+	"path/filepath"
+	"testing"
+
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewFs(t *testing.T) {
+	t.Parallel()
+
+	backendFs := afero.NewOsFs()
+	fs := NewFs(backendFs)
+	assert.Equal(t, backendFs, fs.fs)
+}
+
+func TestOpenLocalFile(t *testing.T) {
+	t.Parallel()
+
+	filename := "github.com/anz-bank/sysl/demo/examples/Modules/deps.sysl"
+	fs := NewFs(afero.NewOsFs())
+	f, err := fs.Open(filename)
+	assert.Nil(t, err)
+	assert.Equal(t, "deps.sysl", filepath.Base(f.Name()))
+}
+
+func TestOpenRemoteFile(t *testing.T) {
+	t.Parallel()
+
+	filename := "github.com/ChloePlanet/sysltestpub/pubbanana.sysl"
+	fs := NewFs(afero.NewOsFs())
```

Comment:
tests should never touch the host filesystem, if it needs to write it should be writing to a MemFs()